### ISSUE TITLE
Simulation in OMEdit crashes when assert is triggert

### ISSUE THREAD

**Issue by niklwors** [OpenModelica/OpenModelica#12407]
### Description

For a model this error for an assert is printed:
```
LOG_ASSERT        | error   | No event found, but assert was triggered. Throwing now!
```
The simulation crashes and the simulation progress bar hangs:
```
Simulation process failed. Exited with code 0xffffffffffffffff.
```

### Expected Behavior

The simulation should not crash and the progress bar should not get stuck. Only an error message should be displayed.

### Version and OS

 - OpenModelica Version: v1.24.0-dev-31-gedfa9cd10e
 - OS: Windows 10, 64 bit

**Comment by adeas31** [comment]
Can you please provide a test model.

**Comment by casella** [comment]
@niklwors I guess this is the same as #11905. In #8200 @AnHeuermann proposed a solution, but then the thing got stuck at some point.

@AnHeuermann would you like to resolve it? That would be great!

**Comment by niklwors** [comment]
Yes, it is the same problem. I had forgotten that I had reported it before. It's just happened again now.

**Comment by niklwors** [comment]
<img width="1031" alt="image002" src="https://github.com/OpenModelica/OpenModelica/assets/12269776/23898be8-7a17-4939-8f87-71a97853eef7">

The problem is that the progress bar hangs and the simulation cannot be aborted

**Comment by casella** [comment]
> The problem is that the progress bar hangs

This is not a bug, this is a feature 😄 The progress bar stops where the simulation aborts, at 54% of the simulation time interval (StartTime->StopTime). In the past it said 100% but that was definitely misleading. Do you find this confusing?

> and the simulation cannot be aborted

The simulation aborts when the red message is issued, hence you cannot abort it again 😃 

I think the behaviour is OK, it's just a matter of improving the text of error messages, because "no event found but assert was triggered. Throwing now!" is not particularly clear to an OMEdit end user

That was the idea of #8200, we just need to get that through.

**Comment by niklwors** [comment]
> > The problem is that the progress bar hangs
> 
> This is not a bug, this is a feature 😄 The progress bar stops where the simulation aborts, at 54% of the simulation time interval (StartTime->StopTime). In the past it said 100% but that was definitely misleading. Do you find this confusing?

Yes, it is a bit confusing. Because you think  the simulation hangs.

> > and the simulation cannot be aborted
> 
> The simulation aborts when the red message is issued, hence you cannot abort it again 😃
> 
> I think the behaviour is OK, it's just a matter of improving the text of error messages, because "no event found but assert was triggered. Throwing now!" is not particularly clear to an OMEdit end user
> 
In out case we do not see the error message. It was therefore not possible to see what was wrong with the model. Only when you start the simulation via a script you could you see the reason for the assert.


> That was the idea of #8200, we just need to get that through.

**Comment by niklwors** [comment]
This is the output windows for our model:
<img width="1005" alt="image001" src="https://github.com/OpenModelica/OpenModelica/assets/12269776/9f279dbb-9f44-471a-b7ec-f71a21ce70ff">

**Comment by perost** [comment]
> This is not a bug, this is a feature 😄 The progress bar stops where the simulation aborts, at 54% of the simulation time interval (StartTime->StopTime). In the past it said 100% but that was definitely misleading. Do you find this confusing?

I also think it's a bit confusing, I can't think of any other example where a progress bar just stops like that. So most users will expect that progress is happening as long as the progress bar is visible.

But also, is the progress bar in the screenshot animated? It kind of looks like it since it's not a uniform green color. I guess it might be different depending on the theme used, but an animated progress bar would be much worse since motion definitely implies that something is happening.

**Comment by casella** [comment]
The fact that the progress bar stops before reaching 100% was implemented in #11128. I am convinced this is the correct approach: having a simulation failing somewhere between startTime and stopTime and yet seeing a green 100% bar is _really_ confusing.

In that ticket, I asked @adeas31 if we could have a red bar in case the simulation was aborted, but it seems like available QT primitives do not allow that. Ditto for the animation. I agree that seeing the bar glowing when the simulation has ended prematurely is bad, but so is seeing all green and 100% success. @adeas31 maybe there is indeed some way to at least stop the glowing thing?

In any case, the point is one should get a clear error in the message browser. If a simulation finishes successfully you see the message "The simulation finished successfully.". If it doesn't, and it aborts prematurely, you should get a message that clearly explains why. "Simulation process failed. Exited with code 0xffffffffffffffff" is not good enough. 

That was the idea of #8200, but only for the case of assert violations. In fact, regarding @niklwors's [reported test case](https://github.com/OpenModelica/OpenModelica/issues/12407#issuecomment-2109528859), it is not really clear why the simulation failed. This is another issue, maybe you can share the model privately so we can run it with the debugger, figure out what is going wrong, and add relevant error messages?

**Comment by bilderbuchi** [comment]
You could modify the progressbar text to "ABORTED at 53%" when the simulation was aborted? That would at least make it a little clearer, although it does not get rid of the animation.

**Comment by ceraolo** [comment]
> In that ticket, I asked @adeas31 if we could have a red bar in case the simulation was aborted, but it seems like available QT primitives do not allow that. 
 If the progress bar color cannot be changed, it can just be substituted by a red rectangle I suppose

**Comment by casella** [comment]
> The error message is in the assert call of the model. But it is not displayed in the output window because of the many warnings. 

I am not sure of that. Further _warnings_ are suppressed, but an assert that terminates the simulation should of course be an _error_, and thus should be displayed. I'm not sure it is, though. @AnHeuermann?

**Comment by niklwors** [comment]
If I disable the log setting LOG_STDOUT, the assert message is displayed in the output window.

**Comment by casella** [comment]
@adrpo, @AnHeuermann I'm always confused about which output streams are used. Can you make sense out of this?
